From 64c11efeeab79ce30fd97e8060af47091d4fe870 Mon Sep 17 00:00:00 2001
From: "Yi, Hongyu" <hongyu.yi@intel.com>
Date: Wed, 10 Jun 2015 14:07:54 +0800
Subject: [PATCH 429/462] Revert "dw9718: Reduce click sounds during power on
 and off"

This reverts commit 4b142d5480569b664f2e04d93f8601620645ae7c.
Since that commit makes VCM power on taking more time, impacting
launch to preview KPI.

Change-Id: Ia7054f78541faf7f76ddebe8d7dca0c00ca8a91e
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-38195
Signed-off-by: Yi, Hongyu <hongyu.yi@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/60139
Build: IMIN (SC)
Approver: Yunliang Ding <yunliang.ding@intel.com>
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Reviewed-by: DaleiX Liu <daleix.liu@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/i2c/imx/dw9718.c          | 36 ++--------------------
 .../camera/drivers/media/i2c/imx/dw9718.h          |  5 +--
 2 files changed, 3 insertions(+), 38 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.c b/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.c
index 6252843..74d5a44 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.c
@@ -101,8 +101,6 @@ int dw9718_vcm_power_up(struct v4l2_subdev *sd)
 	struct i2c_client *client = v4l2_get_subdevdata(sd);
 	int ret;
 	u8 value;
-	int i;
-	int step = DW9718_CLICK_REDUCTION_STEP;
 
 	if (dw9718_dev.power_on)
 		return 0;
@@ -114,8 +112,8 @@ int dw9718_vcm_power_up(struct v4l2_subdev *sd)
 	ret = dw9718_i2c_wr8(client, DW9718_PD, 0);
 	if (ret < 0)
 		goto fail_powerdown;
-	/* Wait t_OPR for VBAT to stabilize */
-	usleep_range(100, 110);
+	/* Wait for VBAT to stabilize */
+	udelay(100);
 
 	/* Detect device */
 	ret = dw9718_i2c_rd8(client, DW9718_SACT, &value);
@@ -140,25 +138,9 @@ int dw9718_vcm_power_up(struct v4l2_subdev *sd)
 	if (ret < 0)
 		goto fail_powerdown;
 
-	/* Wait t_MODE after changing from switching to linear mode */
-	usleep_range(85, 95);
-
-	/* Minimize the click sounds from the lens during power up */
-	i = DW9718_LENS_MOVE_POSITION;
-	while (i <= dw9718_dev.focus) {
-		ret = dw9718_i2c_wr16(client, DW9718_DATA_M, i);
-		if (ret) {
-			dev_err(&client->dev, "%s: write failed\n", __func__);
-			break;
-		}
-		msleep(DW9718_CLICK_REDUCTION_SLEEP);
-		i += step;
-	}
-
 	ret = dw9718_t_focus_abs(sd, dw9718_dev.focus);
 	if (ret)
 		return ret;
-
 	dw9718_dev.initialized = true;
 	dw9718_dev.power_on = 1;
 
@@ -174,24 +156,10 @@ int dw9718_vcm_power_down(struct v4l2_subdev *sd)
 {
 	struct i2c_client *client = v4l2_get_subdevdata(sd);
 	int ret;
-	int i;
-	int step = DW9718_CLICK_REDUCTION_STEP;
 
 	if (!dw9718_dev.power_on)
 		return 0;
 
-	/* Minimize the click sounds from the lens during power down */
-	i = min(dw9718_dev.focus, DW9718_DEFAULT_FOCUS_POSITION) - step;
-	while (i >= DW9718_LENS_MOVE_POSITION) {
-		ret = dw9718_i2c_wr16(client, DW9718_DATA_M, i);
-		if (ret) {
-			dev_err(&client->dev, "%s: write failed\n", __func__);
-			break;
-		}
-		msleep(DW9718_CLICK_REDUCTION_SLEEP);
-		i -= step;
-	}
-
 	ret = dw9718_i2c_wr8(client, DW9718_PD, 1);
 	if (ret)
 		dev_err(&client->dev, "%s: write DW9718_PD to 1 failed\n",
diff --git a/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.h b/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.h
index 2726c7b..fe1fb8a 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.h
@@ -59,9 +59,6 @@ struct dw9718_device {
 #define DW9718_SACT_PERIOD_8_8MS	0x19
 #define DW9718_SACT_DEFAULT_VAL		0x60
 
-#define DW9718_CLICK_REDUCTION_STEP	 30 /* in vcm units */
-#define DW9718_CLICK_REDUCTION_SLEEP	 20 /* in milliseconds */
-#define DW9718_LENS_MOVE_POSITION	120 /* in vcm units */
-#define DW9718_DEFAULT_FOCUS_POSITION	300 /* in vcm units */
+#define DW9718_DEFAULT_FOCUS_POSITION	300
 
 #endif /* __DW9718_H__ */
-- 
1.9.1

